File: backend/app/schemas/attachment.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.schemas.user import UserRead


class AttachmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    task_id: UUID
    uploaded_by: UUID
    uploader: UserRead
    original_filename: str
    content_type: str
    size_bytes: int
    created_at: datetime

File: frontend/src/types/attachment.ts
import type { User } from "@/types/user"

export type Attachment = {
  id: string
  task_id: string
  uploaded_by: string
  uploader: User
  original_filename: string
  content_type: string
  size_bytes: number
  created_at: string
}

export type AttachmentListResponse = {
  items: Attachment[]
}
